### PR TITLE
Make gauss_2d/3d more consistent

### LINKED ISSUE #700: Gaussian helper function cleanup/completeness
In #616 `gaussian_3d` was extended to admit an `indexing` arg.  We may in the future want this for 2d case as well. or just for completeness.

Another improvement we can do there is to admit any iterable of length 3 instead of forcing a `tuple`.  Mainly to support taking in an array.  That would make it easier to use in a loop, ie call in loop over some (k,3) array etc.  Similarly len 2 for 2d.

I think it is a matter of checking if something is an iterable instead of tuple and checking len otherwise assuming scalar.

### PR DESCRIPTION
**Pull request by garrettwrong**
Closes #700 

@j-c-c , Can you double check the unit test change (probably will require looking at it).

I think it is more consistent with 3d, but is all a little suspect.

### CODE REVIEW

**Comment by garrettwrong** [comment]
Summarizing the discussion from yesterday.  It looks as though the 'zyx' mode for gaussian 3d is currently not working and that fixing both 2d and 3d requires swapping x and y.  I'll push a commit that does this and then @j-c-c is going to take over the PR to cover the testing.

**Comment by j-c-c** [comment]
@garrettwrong I cleaned up the tests for 2D and 3D to have (hopefully) clearer labelling for indices. Also, visually confirmed that we are in fact generating Gaussians with the indicated `indexing`!

Ready for your eyes once CI passes.

**Comment by garrettwrong** [comment]
Cool, thanks. I think I'm good with it, but since a lot was my changes we should send to Chris.  Glad we went through it though.

File: src/aspire/utils/misc.py [hunk 1]
```diff
@@ -171,26 +171,37 @@ def gaussian_2d(size, mu=(0, 0), sigma=(1, 1), dtype=np.float64):
     in a square 2D numpy array.
 
     :param size: The length of each dimension of the returned array (pixels)
-    :param mu: A 2-tuple, :math:`(\\mu_x, \\mu_y)`, indicating the center of the Gaussian
-    :param sigma: A 2-tuple, :math:`(\\sigma_x, \\sigma_y)`, of the standard
+    :param mu: Iterable of len(2), :math:`(\\mu_x, \\mu_y)`, indicating the center of the Gaussian
+    :param sigma: Iterable of len(2) or constant, :math:`(\\sigma_x, \\sigma_y)`, of the standard
             deviation in the x and y directions. A single value, :math:`\\sigma`, can be
             used when :math:`\\sigma_x = \\sigma_y`.
+    :param indexing: The order of axis indexing, passed to grid_2d
```
Comment (chris-langfield) [inline src/aspire/utils/misc.py]: General question: I wonder if we should give the full module "path" to functions like this when we reference them. So e.g. `aspire.utils.grid_2d`?

File: src/aspire/utils/misc.py [hunk 2]
```diff
@@ -206,33 +217,39 @@ def gaussian_3d(size, mu=(0, 0, 0), sigma=(1, 1, 1), indexing="zyx", dtype=np.fl
...
-    else:
-        assert (
-            isinstance(sigma, tuple) and len(sigma) == 3
-        ), "sigma must be a scalar or 3-tuple."
+
+    mu = np.array(mu, dtype=dtype)
+    sigma = np.array(sigma, dtype=dtype)
+    if len(mu) != 3:
+        raise ValueError("`mu` must be len(3).")
+    if len(sigma) != 3:
```
Comment (chris-langfield) [inline src/aspire/utils/misc.py:184]: I would just throw in some tests to cover these checks, e.g. a length n-1 mu and a length n+1 sigma

**Review by chris-langfield — CHANGES_REQUESTED** [review]

File: src/aspire/utils/misc.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/utils/misc.py]: Wouldn't hurt.

Comment (j-c-c) [reply src/aspire/utils/misc.py]: Done. Will keep in mind in the future

File: src/aspire/utils/misc.py (hunk 2 above)
Comment (j-c-c) [reply src/aspire/utils/misc.py:184]: Good call. Done.

**Review by chris-langfield — APPROVED** [review]
Looks good thanks!

**Comment by garrettwrong** [comment]
LGTM too, thanks Josh! If no objections I'll merge this afternoon.

**Comment by j-c-c** [comment]
Merge away!

**Comment by garrettwrong** [comment]
Will merge after CI completes.

**Pull request merged**